--- /etc/passwd 2018-01-10 10:03:52.000000000 +0100
+++ /etc/passwd 2018-01-18 22:08:48.000000000 +0100
@@ -1,6 +1,8 @@
-root:x:0:0:root:/root:/bin/ash
+root:x:0:0:root:/tmp:/bin/ash
 daemon:*:1:1:daemon:/var:/bin/false
 ftp:*:55:55:ftp:/home/ftp:/bin/false
 network:*:101:101:network:/var:/bin/false
-nobody:*:65534:65534:nobody:/var:/bin/false
 dnsmasq:x:453:453:dnsmasq:/var/run/dnsmasq:/bin/false
+bird:x:479:479:bird:/var:/bin/false
+collectd:x:499:499:collectd:/tmp:/bin/false
+nobody:*:65534:65534:nobody:/var:/bin/false
