--- /etc/init.d/bird6 2018-01-10 10:03:52.000000000 +0100
+++ /etc/init.d/bird6 2018-01-20 15:16:28.000000000 +0100
@@ -6,21 +6,21 @@
 START=99
 STOP=10
 
-SERVICE_DAEMONIZE=1
-SERVICE_USE_PID=1
+USE_PROCD=1
 SERVICE_PID_FILE="/var/run/$BIRD.pid"
 
 BIRD_BIN="/usr/sbin/$BIRD"
 BIRD_CONF="/etc/$BIRD.conf"
 
-start() {
-	service_start $BIRD_BIN -d -c $BIRD_CONF -P $SERVICE_PID_FILE
-}
+USER=bird
 
-stop() {
-	service_stop $BIRD_BIN
-}
+start_service() {
+  : >> "$SERVICE_PID_FILE"
+  chown bird:bird "$SERVICE_PID_FILE"
 
-reload() {
-	service_reload $BIRD_BIN
+  procd_open_instance
+  procd_set_param command "$BIRD_BIN"
+  procd_append_param command -f -c $BIRD_CONF -P $SERVICE_PID_FILE -u $USER
+  procd_set_param respawn
+  procd_close_instance
 }
